iodata/iodata.py

```diff
@@ -259,22 +260,31 @@ def atcorenums(self) -> np.ndarray:
             # https://github.com/PyCQA/pylint/issues/1694
             # pylint: disable=no-member
             result = self.atnums.astype(float)
-            self._atcorenums = result
+            self.atcorenums = result
         return result
 
     @atcorenums.setter
     def atcorenums(self, atcorenums):
         self._atcorenums = atcorenums
+        if self._charge is not None:
+            # _charge is treated as the dependent one, while atcorenums and
+            # nelec are treated as independent.
+            if self.nelec is None:
```

Comment:
```suggestion
            if self.nelec is None and atcorenums is not None:
```
Maybe this fix `AttributeError: 'NoneType' object has no attribute 'sum'`

Reply:
I'll check

Reply:
To get intuitive behavior, the code had to be extended, which is now the case in the latest commit. I've added several variations of unit tests to make sure it works (and checked with coverage that they reached all code paths.)